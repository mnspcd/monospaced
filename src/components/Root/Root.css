@import url("../../system/fonts/index.css");

*,
::after,
::before {
  box-sizing: border-box;
}

html {
  background-color: var(--brand-primary-light-color);
  color: var(--text-color);
  font-family: var(--font-body-text);
  font-variant: common-ligatures contextual;
  line-height: var(--leading);
  text-rendering: optimizeLegibility;
}

body {
  margin: 0;
}

@media (--larger-viewport) {
  body {
    font-size: 1.25rem;
  }
}

@media (--largest-viewport) {
  body {
    font-size: 1.5rem;
  }
}

a {
  color: var(--mnspcd-color-blue-6);
  text-decoration: none;
}

a:hover {
  color: var(--mnspcd-color-blue-7);
  outline: 0;
}

a:active {
  color: var(--mnspcd-color-blue-8);
  outline: 0;
}

b,
h1,
h2,
h3,
h4,
h5,
h6,
strong,
th {
  font-weight: 400; /* never bold */
}

code {
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-s));
  font-variant: slashed-zero;
}

pre {
  font-family: var(--font-monospaced);
  margin: 0;
}

small {
  font-size: calc(1em * var(--type-scale-m));
}

@media (--larger-viewport) {
  small {
    font-size: calc(1em * var(--type-scale-s));
  }
}

sub {
  font-variant-position: sub;
}

sup {
  font-variant-position: super;
}

time {
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-m));
  font-variant: slashed-zero;
}

@media (--larger-viewport) {
  time {
    font-size: calc(1em * var(--type-scale-s));
  }
}

.Root {
  background-color: var(--background-color);
}

.u-monospaced {
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
}

.u-smallCaps {
  font-family: var(--font-small-caps);
  text-transform: lowercase;
}
