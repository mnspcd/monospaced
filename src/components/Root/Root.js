import PropTypes from "prop-types";
import React from "react";

import "../../assets/.htaccess";
import "./Root.css";
import Head from "./Head";
import { version } from "../../../package.json";

const Root = ({ meta, router }) => {
  return (
    <html lang="en-gb">
      <Head meta={meta} />
      <body className="no-js">
        <script
          dangerouslySetInnerHTML={{ __html: `document.body.className = ""` }}
        />
        <div className="Root" id="router" dangerouslySetInnerHTML={router} />
        <script src={`/bundle.js?v=${version}`} />
        <script
          dangerouslySetInnerHTML={{
            __html: `var c = document.createElement("link");c.rel = "stylesheet";c.href = "//hello.myfonts.net/count/37bce3";var h = document.head;n = h.childNodes;h.insertBefore(c, n[n.length - 1 ].nextSibling);`,
          }}
        />
      </body>
    </html>
  );
};

Root.propTypes = {
  meta: PropTypes.array,
  router: PropTypes.array,
};

export default Root;
