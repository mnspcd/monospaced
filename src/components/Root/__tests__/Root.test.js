/* eslint-env jest */
import React from "react";
import { shallow } from "enzyme";

import Root from "../Root";

jest.mock("../../../../package.json", () => ({ version: "0.0.0" }));

describe("Root component", () => {
  it("should render correctly", () => {
    const component = shallow(<Root meta={[]} router={[]} />);
    expect(component).toMatchSnapshot();
  });
});
