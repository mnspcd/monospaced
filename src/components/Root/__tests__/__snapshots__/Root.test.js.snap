// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should render correctly 1`] = `
<html
  lang="en-gb"
>
  <Head
    meta={Array []}
  />
  <body
    className="no-js"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "document.body.className = \\"\\"",
        }
      }
    />
    <div
      className="Root"
      dangerouslySetInnerHTML={<div />}
      id="router"
    />
    <script
      src="/bundle.js?v=0.0.0"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "var c = document.createElement(\\"link\\");c.rel = \\"stylesheet\\";c.href = \\"//hello.myfonts.net/count/37bce3\\";var h = document.head;n = h.childNodes;h.insertBefore(c, n[n.length - 1 ].nextSibling);",
        }
      }
    />
  </body>
</html>
`;
